File: 自定义键盘/CustomKeyBoard/CustomKeyBoard/ViewController.m
//
//  ViewController.m
//  CustomKeyBoard
//
//  Created by cp316 on 17/3/2.
//  Copyright © 2017年 lanht. All rights reserved.
//

#import "ViewController.h"
#import "KeyboardView.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *inputTf;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    KeyboardView *kbView = [[KeyboardView alloc]init];
    kbView.inputSource = self.inputTf;
    self.inputTf.inputView = kbView;
}

@end

File: Swift/懒加载/懒加载/ViewController.swift
 //
//  ViewController.swift
//  懒加载
//
//  Created by cp316 on 17/3/6.
//  Copyright © 2017年 lanht. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    // 简单写法
    lazy var label: DemoLabel = DemoLabel()

    // 完整写法 日常开发中用上面简写法
    lazy var label2: DemoLabel = { ()->DemoLabel in
        let l = DemoLabel()
        return l
    }()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
//        view.addSubview(label)
        
        
    }

}

